controller/share_manager_controller.go

```diff
@@ -330,11 +378,11 @@ func (c *ShareManagerController) syncShareManager(key string) (err error) {
 		}
 	}()
 
-	if err = c.syncShareManagerVolume(sm); err != nil {
+	if err = c.syncShareManagerPod(sm); err != nil {
 		return err
 	}
 
-	if err = c.syncShareManagerPod(sm); err != nil {
+	if err = c.syncShareManagerVolume(sm); err != nil {
```

Comment:
Not sure either. Can you help to explain this one @james-munson ?

Reply:
At some point early in the development and testing, I thought it made a difference.  And it sort of does, because Pod sync will be repeated until it is error-free before any volume sync is attempted.  But in general, I figured it probably does not.
Is there a reason that it should be re-reversed?  That volume should in fact be sync'ed first?

Reply:
> Is there a reason that it should be re-reversed? That volume should in fact be sync'ed first?

The natural flow is volume should attached/detact first before we create/delete a pod so the previous logic of sync volume first then sm pod make more sense. Unless it is required, maybe we shouldn't change it?

Reply:
Will the volume know that it should be detached?  All the delinquency logic is in the pod sync.  But I take your point that this might perturb the order for RWO volumes.
We can swap back, and do the testing to ensure that the RWX sequence still works.

Reply:
> Will the volume know that it should be detached? All the delinquency logic is in the pod sync. 

Maybe it will require the second run of sync loop to have volume detach